compiler/lib/global_flow.ml

```diff
@@ -243,14 +243,9 @@ let program_deps st { blocks; _ } =
       | Cond (x, cont1, cont2) ->
           cont_deps blocks st cont1;
           cont_deps blocks st ~ignore:x cont2
-      | Switch (x, a1, a2) ->
+      | Switch (x, a1) ->
           Array.iter a1 ~f:(fun cont -> cont_deps blocks st cont);
-          Array.iter a2 ~f:(fun cont -> cont_deps blocks st cont);
           let h = Hashtbl.create 16 in
-          Array.iteri
-            ~f:(fun i (pc, _) ->
-              Hashtbl.replace h pc (i :: (try Hashtbl.find h pc with Not_found -> [])))
```

Comment:
@vouillon, how important is the optimization about known_cases ? do we need to restore it ?

Reply:
I've tried to restore the optimization but it appears that our test-suite doesn't capture/test this.
Would you be able to contribute an expect test for this ?

Reply:
I'm not sure it is really useful in practice. It makes the analysis more precise and does not cost much, but it's not clear to me whether it make much of a difference.

Reply:
Your change looks fine. I need to think of a test.